wolfcrypt/src/asn.c

```diff
@@ -14384,12 +14400,45 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
                 ret = ASN_PARSE_E;
         }
         if (ret == 0) {
-            ret = ASNToHexString(input, inOutIdx, (char**)&curve->Af, inSz,
+            char* af = NULL;
+            ret = ASNToHexString(input, inOutIdx, &af, inSz,
                                             key->heap, DYNAMIC_TYPE_ECC_BUFFER);
+        #ifndef WOLFSSL_ECC_CURVE_STATIC
```

Comment:
This code pattern looks like it's used in 4 places.  Can it be isolated for ease of use and maintenance?

Reply:
Fixed. Thanks Todd